src/editor/appmapEditorProvider.ts

```
@@ -103,31 +105,70 @@ export default class AppMapEditorProvider
     private readonly extensionState: ExtensionState
   ) {}
 
-  async openCustomDocument(uri: vscode.Uri): Promise<AppMapDocument> {
-    const appmapFileData = fs.statSync(uri.fsPath);
-    const appmapFileSize = appmapFileData.size;
-    const functions = await this.generateStats(uri);
+  async openCustomDocument(appMapOrSequenceDiagramDiffUri: vscode.Uri): Promise<AppMapDocument> {
+    let appMapUri: vscode.Uri;
+    let sequenceDiagramData: string | undefined;
+
+    function abortSequenceDiagramDiff(msg: string): AppMapDocument {
+      vscode.window.showInformationMessage(['Unable to open sequence diagram', msg].join(': '));
+      // Give ourselves a way to short circuit this openCustomDocument method.
+      return undefined as unknown as AppMapDocument;
+    }
+
+    if (appMapOrSequenceDiagramDiffUri.fsPath.endsWith('.diff.sequence.json')) {
+      sequenceDiagramData = await readFile(appMapOrSequenceDiagramDiffUri.fsPath, 'utf-8');
+      const appMapTokens = appMapOrSequenceDiagramDiffUri.fsPath.split('/');
+      const diffIndex = appMapTokens.lastIndexOf('diff');
+      if (diffIndex === -1)
+        return abortSequenceDiagramDiff(
+          `Expected to find a 'diff' directory in the path ${appMapOrSequenceDiagramDiffUri.fsPath}`
+        );
```

This violates the `CustomReadonlyEditorProvider` interface. What happens when `undefined` is returned? It'd be better to raise an exception otherwise this may result in unexpected behavior in the future.